src/linearmixedmodel.jl

```diff
@@ -957,29 +963,29 @@ Update the blocked lower Cholesky factor, `m.L`, from `m.A` and `m.reterms` (use
 This is the crucial step in evaluating the objective, given a new parameter value.
 """
 function updateL!(m::LinearMixedModel{T}) where {T}
-    A = m.A
-    L = m.L
-    k = length(m.allterms)
+    A, L, reterms = m.A, m.L, m.reterms
+    k = length(reterms)
     for (l, a) in zip(L, A) # copy each element of A to corresponding element of L
         copyto!(l, a)       # For some reason copyto!.(L, A) allocates a lot of memory
     end
-    for (j, cj) in enumerate(m.reterms)  # pre- and post-multiply by Λ, add I to diagonal
+    for j in eachindex(reterms) # pre- and post-multiply by Λ, add I to diagonal
+        cj = reterms[j]
         scaleinflate!(L[kp1choose2(j)], cj)
-        for i = (j+1):k         # postmultiply column by Λ
+        for i = (j+1):(k+1)     # postmultiply column by Λ
```

Comment:
`k+1` because `k` is now just the number of `reterms` and we need to add one for the `feterm`, right?

Reply:
Yes, I believe you are correct.  Good that you caught that.